install/helm/agones/templates/service/allocation.yaml

```
@@ -72,6 +73,9 @@ spec:
       protocol: TCP
 {{- end }}
   type: {{ .Values.agones.allocator.service.serviceType }}
+{{- if $clusterIP }}
```

```suggestion
{{- if (ne .Values.agones.allocator.service.clusterIP "") }}
```

I write this because this change sent me lookinga thte K8s [reference docs](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/#:~:text=to%20type%20ExternalName.-,clusterIP%20(string),-clusterIP%20is%20the), and looking at `Service.spec.clusterIP`, and I saw it say:

> Only applies to types ClusterIP, NodePort, and LoadBalancer

So it's not just the `ClusterIP` value for `Service.spec.type` -- so I'd recommend leaving it open, and let the end user decide when it's appropriate to set the value, rather than constrain them?